src/components/task/existing-task/edit-task/index.jsx

```diff
@@ -0,0 +1,83 @@
+import React, { useEffect } from "react";
+import propTypes from "prop-types";
+import { useForm } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { useDispatch } from "react-redux";
+import TextArea from "components/base/text-area";
+import Button from "components/base/button";
+import Image from "components/base/image";
+import {
+  TASK_TEXTAREA_NUM_OF_ROW,
+  TYPE_BUTTON,
+  TITLE_FIELD_NAME_ATTRIBUTE,
+  PATH_CHECK_ICON,
+  ALT_CHECK_ICON_TAG,
+  PATH_DELETE_ICON,
+  ALT_DELETE_ICON_TAG,
+} from "utils/const";
+import { taskSchema } from "utils/schema";
+import { editTask, markAsDone } from "store/actions/todo";
+
+function EditTaskForm({ taskId, existingTitle, onDelete, onTaskEdit }) {
+  const dispath = useDispatch();
+  useEffect(
+    () => setValue(TITLE_FIELD_NAME_ATTRIBUTE, existingTitle),
+    [existingTitle]
+  );
+
+  const updateTask = (task) => {
+    dispath(editTask({ taskId, title: task.title }));
+    setValue(TITLE_FIELD_NAME_ATTRIBUTE, null);
+    onTaskEdit();
+  };
+
+  const saveAsDone = (task) => {
+    dispath(editTask({ taskId, title: task.title }));
+    dispath(markAsDone(taskId));
+    setValue(TITLE_FIELD_NAME_ATTRIBUTE, null);
+    onTaskEdit();
+  };
+
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    setFocus,
+    formState: { errors },
+  } = useForm({
+    mode: "onChange",
+    resolver: yupResolver(taskSchema),
+  });
+
+  useEffect(() => {
+    setFocus(TITLE_FIELD_NAME_ATTRIBUTE);
+  }, [setFocus]);
```

Comment:
reference to the function `setFocus` will probably not going to be changed hence using it as a dependency is not useful.

```suggestion
  useEffect(() => {
    setFocus(TITLE_FIELD_NAME_ATTRIBUTE);
  }, []);
```